cli/cmd/runtime/start.go

```diff
@@ -32,19 +32,20 @@ import (
 // Env var keys must be prefixed with RILL_RUNTIME_ and are converted from snake_case to CamelCase.
 // For example RILL_RUNTIME_HTTP_PORT is mapped to Config.HTTPPort.
 type Config struct {
-	HTTPPort             int           `default:"8080" split_words:"true"`
-	GRPCPort             int           `default:"9090" split_words:"true"`
-	LogLevel             zapcore.Level `default:"info" split_words:"true"`
-	MetastoreDriver      string        `default:"sqlite"`
-	MetastoreURL         string        `default:"file:rill?mode=memory&cache=shared" split_words:"true"`
-	AllowedOrigins       []string      `default:"*" split_words:"true"`
-	AuthEnable           bool          `default:"false" split_words:"true"`
-	AuthIssuerURL        string        `default:"" split_words:"true"`
-	AuthAudienceURL      string        `default:"" split_words:"true"`
-	SafeSourceRefresh    bool          `default:"false" split_words:"true"`
-	ConnectionCacheSize  int           `default:"100" split_words:"true"`
-	QueryCacheSize       int           `default:"10000" split_words:"true"`
-	AllowHostCredentials bool          `default:"false" split_words:"true"`
+	HTTPPort                    int           `default:"8080" split_words:"true"`
+	GRPCPort                    int           `default:"9090" split_words:"true"`
+	LogLevel                    zapcore.Level `default:"info" split_words:"true"`
+	MetastoreDriver             string        `default:"sqlite"`
+	MetastoreURL                string        `default:"file:rill?mode=memory&cache=shared" split_words:"true"`
+	AllowedOrigins              []string      `default:"*" split_words:"true"`
+	AuthEnable                  bool          `default:"false" split_words:"true"`
+	AuthIssuerURL               string        `default:"" split_words:"true"`
+	AuthAudienceURL             string        `default:"" split_words:"true"`
+	SafeSourceRefresh           bool          `default:"false" split_words:"true"`
+	ConnectionCacheSize         int           `default:"100" split_words:"true"`
+	QueryCacheSize              int           `default:"10000" split_words:"true"`
+	AllowHostCredentials        bool          `default:"false" split_words:"true"`
+	InstanceStorageLimitInBytes int64         `default:"10737418240" split_words:"true"` // 10GB
```

Comment:
> This will need to be configurable on an instance-level. Since there may be different tiers, e.g. where pricing is different for a 10GB and 50GB project.

Sure I imagined that we would keep a default at runtime level with ability to modify it at instance level. Do we want to keep it configurable as part of this PR itself ? Do we want to keep the default or remove it completely?

> Also, we can't really guarantee the storage limit here, so maybe it should be IngestionLimitBytes instead?

Yeah this makes sense.

Reply:
No need for a runtime-level default, just setting on instance is fine. If it's set to 0 on the instance, it should mean no limit.